google-cloud-storage/src/main/java/com/google/cloud/storage/Channel.java

```diff
@@ -0,0 +1,294 @@
+    this.kind = builder.kind;
+    this.id = builder.id;
+    this.resourceId = builder.resourceId;
+    this.resourceUri = builder.resourceUri;
+    this.token = builder.token;
+    this.expiration = builder.expiration;
+    this.type = builder.type;
+    this.address = builder.address;
+    this.params = builder.params;
+    this.payload = builder.payload;
+  }
+
+  /* Identifies this as a notification channel used to watch for changes to a resource*/
+  public String getKind() {
+    return kind;
+  }
+
+  /* Returns a UUID or similar unique string that identifies this channel.*/
+  public String getId() {
+    return id;
+  }
+
+  /* Returns an opaque ID that identifies the resource being watched on this channel.*/
+  public String getResourceId() {
+    return resourceId;
+  }
+
+  /* Returns a version-specific identifier for the watched resource.*/
+  public String getResourceUri() {
+    return resourceUri;
+  }
+
+  /* Returns an arbitrary string delivered to the target address with each notification delivered over the channel.*/
+  public String getToken() {
+    return token;
+  }
+
+  /* Returns the notification channel expiration date and time.*/
+  public Long getExpiration() {
+    return expiration;
+  }
+
+  /* Returns the type of delivery mechanism used for this channel.*/
+  public String getType() {
+    return type;
+  }
+
+  /* Returns the address where notifications are delivered.*/
+  public String getAddress() {
+    return address;
+  }
+
+  /* Returns the newly declared parameter by name.*/
+  public Map<String, String> getParams() {
+    return params;
+  }
+
+  /* Returns a Boolean value to indicate whether payload is wanted.*/
+  public Boolean getPayload() {
+    return payload;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Channel channel = (Channel) o;
+    return Objects.equals(kind, channel.kind)
+        && Objects.equals(id, channel.id)
+        && Objects.equals(resourceId, channel.resourceId)
+        && Objects.equals(resourceUri, channel.resourceUri)
+        && Objects.equals(token, channel.token)
+        && Objects.equals(expiration, channel.expiration)
+        && Objects.equals(type, channel.type)
+        && Objects.equals(address, channel.address)
+        && Objects.equals(params, channel.params)
+        && Objects.equals(payload, channel.payload);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        kind, id, resourceId, resourceUri, token, expiration, type, address, params, payload);
+  }
+
+  /** Returns a builder for the {@link Channel} object. */
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /** Returns a builder for the {@link Channel} object. */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  com.google.api.services.storage.model.Channel toProtobuf() {
```

Comment:
I remembered we were using toPb() all over the java clients. Is this a new pattern for the same?

Reply:
yes may be, am also familiar with `toPb(`)  but @Elharo suggest me to use` toProtobubuf() `instead of toPb.